core/lib/src/unix.rs

```diff
@@ -0,0 +1,80 @@
+
+use rocket_http::hyper::net::NetworkStream;
+#[cfg(unix)] use rocket_http::unix::UnixSocketStream;
+#[cfg(all(unix, feature = "tls"))] use rocket_http::unix::UnixHttpsStream;
+use data::NetStream;
+
+macro_rules! check_unix_domain_socket {
```

Comment:
Is `lock_unix_domain_socket` a better name for this?

My understanding is that this all works because `lock_file` is still in scope once Rocket yields to hyper at `handle_threads`. I think this should be documented in a comment somewhere.